diff --git a/organizations/admin/classes/domain/Organization.php b/organizations/admin/classes/domain/Organization.php
index 4e842d8..3fbce82 100644
--- a/organizations/admin/classes/domain/Organization.php
+++ b/organizations/admin/classes/domain/Organization.php
@@ -639,10 +639,10 @@ class Organization extends DatabaseObject {
 		$alphArray = array();
 		$result = mysqli_query($this->db->getDatabase(), "SELECT DISTINCT UPPER(SUBSTR(TRIM(LEADING 'The ' FROM name),1,1)) letter, COUNT(SUBSTR(TRIM(LEADING 'The ' FROM name),1,1)) letter_count
 								FROM Organization O
-								GROUP BY SUBSTR(TRIM(LEADING 'The ' FROM name),1,1)
+								GROUP BY UPPER(SUBSTR(TRIM(LEADING 'The ' FROM name),1,1))
 								ORDER BY 1;");
 
-		while ($row = mysqli_fetch_assoc($result)){
+		while ($row = !empty($result) ? mysqli_fetch_assoc($result) : FALSE) {
 			$alphArray[$row['letter']] = $row['letter_count'];
 		}
 
