diff --git a/auth/admin/classes/common/Utility.php b/auth/admin/classes/common/Utility.php
index 9b9006df..b5698050 100644
--- a/auth/admin/classes/common/Utility.php
+++ b/auth/admin/classes/common/Utility.php
@@ -68,7 +68,9 @@ public function getCORALPath(){
 	//returns page URL up to /coral/
 	public function getCORALURL(){
 		$pageURL = 'http';
-		if ($_SERVER["HTTPS"] == "on") {$pageURL .= "s";}
+    if (isset($_SERVER["HTTPS"]) && $_SERVER["HTTPS"] == "on") {
+      $pageURL .= "s";
+    }
 		$pageURL .= "://";
 		if ($_SERVER["SERVER_PORT"] != "80") {
 		  $pageURL .= $_SERVER["SERVER_NAME"].":".$_SERVER["SERVER_PORT"];
